spec/create_table_statement_spec.cr

```diff
@@ -26,6 +26,47 @@ describe LuckyMigrator::CreateTableStatement do
     SQL
   end
 
+  it "sets default values" do
+    future_time = Time.new(2030,1,1)
+
+    built = LuckyMigrator::CreateTableStatement.new(:users).build do
+      add name : String, default: "name"
+      add email : String?, default: "optional"
+      add age : Int32, default: 1
+      add num : Int64, default: 1
+      add amount_paid : Float, default: 1.0
+      add completed : Bool, default: false
+      add joined_at : Time, default: :now
+      add future_time : Time, default: future_time
+    end
+
+    built.statements.size.should eq 1
+    built.statements.first.should eq <<-SQL
+    CREATE TABLE users (
+      id serial PRIMARY KEY,
+      created_at timestamptz NOT NULL,
+      updated_at timestamptz NOT NULL,
+      name text NOT NULL DEFAULT 'name',
+      email text DEFAULT 'optional',
+      age int NOT NULL DEFAULT 1,
+      num bigint NOT NULL DEFAULT 1,
+      amount_paid decimal NOT NULL DEFAULT 1.0,
+      completed boolean NOT NULL DEFAULT false,
+      joined_at timestamptz NOT NULL DEFAULT NOW(),
+      future_time timestamptz NOT NULL DEFAULT '#{future_time.to_utc}');
+    SQL
+  end
+
+  it "raises on Int32 column with Int64 default" do
+    future_time = Time.new(2030,1,1)
+
+    expect_raises Exception, "Cannot set Int64 default for Int32 column 'age'. Either set the type to Int64 or change the default value." do
+      built = LuckyMigrator::CreateTableStatement.new(:users).build do
+        add age : Int32, default: 2_147_483_648
```

Comment:
You can force something to be an Int64 by doing this: `1_i64` That creates an Int64 with the value of `1`

Reply:
The case I'm trying to catch is when the default is Int64 and the database column is Int32. Converting n Int64 number to Int32 like 3000000000_i32 raises "3000000000 doesn't fit in an Int32"

Reply:
I opted to just remove this overload: `def default_value(type : Int32.class, default : Int64)`